source/common/http/conn_manager_impl.cc

```diff
@@ -470,6 +487,47 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
       connection_manager_.config_, *snapped_route_config_, connection_manager_.random_generator_,
       connection_manager_.runtime_, connection_manager_.local_info_);
 
+  if (!cached_route_.valid()) {
+    cached_route_.value(snapped_route_config_->route(*request_headers_, stream_id_));
+  }
+
+  // Check for WebSocket upgrade request if the route exists, and supports WebSockets.
+  if ((protocol == Protocol::Http11) && cached_route_.value()) {
```

Comment:
Yes and a prior version of the PR actually sent a 404 if no filters were configured. But then, @dnoe pointed to docs that stated that route filter must be configured for HTTP conn man, and given that websocket is under http conn man, I removed it. I can add the check for lack of route filter for websockets but if you want it to be a http conn man wide feature, its probably done in a separate PR.